### ISSUE TITLE
[Accessibility] Clarity datagrid VPAT compliant 

### ISSUE THREAD

**Issue by dkanev** [vmware/clarity#2597]
## Summary

Our product which uses clarity has VPAT assessment report which contains several issues relate to the clarity datagrid.

**Issue 1: The radio buttons on the grids, have labels that are properly associated with the fields, but the LABEL element is empty.**  

**Instance Severity: 10**

![claritydatagrid-radiobuttons](https://user-images.githubusercontent.com/439779/44724557-b9689a00-aadb-11e8-8608-5a242e91273c.png)

**Impact**
When form fields do not explicitly include a label, assistive technologies may incorrectly render the label or provide no label at all to users. When labels are not present or are incorrect, users of assistive technologies may not be able to complete a form.

**Code Reference**
```
<div class="radio">
<input id="clr-dg-row0" class="ng-untouched ng-pristine ng-valid" type="radio">
<label for="clr-dg-row0"></label>
</div>
```

**Propose Solution from VPAT assessment team**
Provide valid textual information within the LABEL element. 

**Code Example**
```
<div class="radio">
<input id="clr-dg-row0" class="ng-untouched ng-pristine ng-valid" type="radio">
<label for="clr-dg-row0">Host Connection</label>
</div>

```

**Issue 2: The only text within each button to expand the row is "angle".** 

**Instance Severity: 10**

![datagrid-expandbutton](https://user-images.githubusercontent.com/439779/44724626-f3d23700-aadb-11e8-85e8-348129cd17dc.png)

**Code Reference**
```
<button class="datagrid-expandable-caret-button ng-star-inserted" type="button">
	<clr-icon class="datagrid-expandable-caret-icon" shape="caret" dir="down">
		<svg version="1.1" viewBox="0 0 36 36" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" focusable="false" role="img">
		<title>angle</title>
		<path class="clr-i-outline clr-i-outline-path-1" d="M29.52,22.52,18,10.6,6.48,22.52a1.7,1.7,0,0,0,2.45,2.36L18,15.49l9.08,9.39a1.7,1.7,0,0,0,2.45-2.36Z"></path>
		</svg>
	</clr-icon>
</button>
```

**Propose Solution from VPAT assessment team**
There are several solutions in this particular case including modifying the title, adding aria-label, and others. In the example below, aria-hidden is added to the graphic and a screen reader only text is put in place. 

**Code Example**
```
<style type="text/css">
.sr-only-clip {
  position: absolute !important;
  height: 1px; 
  width: 1px; 
  overflow: hidden;
  padding: 0 !important;
  border: 0! important;
  white-space: nowrap !important;
  clip: rect(1px 1px 1px 1px) !important; /* IE6, IE7 */
  clip: rect(1px, 1px, 1px, 1px) !important;
  clip-path: inset(50%) !important;
}
</style>

<button class="datagrid-expandable-caret-button ng-star-inserted" type="button">
	<clr-icon class="datagrid-expandable-caret-icon" shape="caret" dir="down">
		<svg [...snip...] role="img" aria-hidden="true">
		[... snip ...]
		</svg>
	</clr-icon>
	<span class="sr-only-clip">Expand Statistics Section</span>
</button>

```

Probably the same issues can apply for filter column button, and buttons in the pagination component

Please evaluate how this issues can be address.

**Comment by youdz** [comment]
Regarding issue 1, we haven't had issues so far because the radio button are obviously associated with each row (since the datagrid is properly read as a table by screenreaders), so a proper title before the datagrid or some other form of explanation as to why you are selecting items has always been enough. We would indeed like a generic "Select" label on the checkboxes, but it is more of a "good to have" and the dev efforts for it are extremely high as I will explain for the next issue.

Also, please note that to have such app-specific labels as your "Host Connection" example, we would have to just provide you with a way to specify the label, and you would be responsible for including labels for every single row in every datagrid of your app.

Issue 2 is a known issue for us. The problem is that the title we need to include for accessibility would need to be internationalized, otherwise it would simply not be accessible for non-English languages.
We have been considering a couple solutions to this problem, but they all require a major overhaul across all of Clarity's components to be shipped, and we simply haven't had the time to address it.

Once again, our solutions would simply label the caret "Expand", any app-specific label would have to be part of the app itself, bloating the API and creating much more integration work for you every time. That's why the simple label seems like a better idea to us.

**Comment by dkanev** [comment]
Hi @youdz ,
thanks for quick replay, what about to be provide default labels on english for the caret, radio buttons, check box and pagination buttons and additionally to be provided a way, these  buttons to be overwrite from the consumers of the grid. Just and idea.

Thanks.

**Comment by mathisscott** [comment]
Hi @dkanev 
Thanks for the suggestion. This is what @youdz is describing in his comment. Doing that replacement you're speaking of is non-trivial. And without it the component could not be internationalized and would not be accessible to non-English speakers.
